{"primaryContentSections":[{"declarations":[{"languages":["swift"],"platforms":["Mac Catalyst"],"tokens":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"NfinityCloudError"}]}],"kind":"declarations"}],"variants":[{"paths":["\/documentation\/passwordandnfinityframework\/nfinityclouderror"],"traits":[{"interfaceLanguage":"swift"}]}],"hierarchy":{"paths":[["doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework"]]},"kind":"symbol","identifier":{"url":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError","interfaceLanguage":"swift"},"topicSections":[{"title":"Enumeration Cases","identifiers":["doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/attemptExceeded","doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/cloudError(errorCode:errorMessage:)","doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/cloudHTTPError(httpCode:response:)","doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/httpDeserialization(httpCode:cause:headers:)","doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/network(cause:)"]},{"identifiers":["doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/Error-Implementations"],"title":"Default Implementations","generated":true}],"abstract":[{"type":"text","text":"Represents an enum hierarchy for errors related to the Nfinity Cloud."}],"sections":[],"metadata":{"symbolKind":"enum","roleHeading":"Enumeration","externalID":"s:27PasswordAndNfinityFramework0C10CloudErrorO","title":"NfinityCloudError","fragments":[{"kind":"keyword","text":"enum"},{"text":" ","kind":"text"},{"text":"NfinityCloudError","kind":"identifier"}],"role":"symbol","navigatorTitle":[{"kind":"identifier","text":"NfinityCloudError"}],"modules":[{"name":"PasswordAndNfinityFramework"}]},"schemaVersion":{"major":0,"patch":0,"minor":3},"relationshipsSections":[{"title":"Conforms To","kind":"relationships","identifiers":["doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/s5ErrorP","doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/s8SendableP"],"type":"conformsTo"}],"references":{"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError/httpDeserialization(httpCode:cause:headers:)":{"identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/httpDeserialization(httpCode:cause:headers:)","type":"topic","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror\/httpdeserialization(httpcode:cause:headers:)","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"text":"httpDeserialization","kind":"identifier"},{"kind":"text","text":"("},{"kind":"externalParam","text":"httpCode"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:Si","kind":"typeIdentifier","text":"Int"},{"kind":"text","text":", "},{"text":"cause","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier","text":"Error"},{"kind":"text","text":", "},{"text":"headers","kind":"externalParam"},{"text":": [","kind":"text"},{"text":"AnyHashable","preciseIdentifier":"s:s11AnyHashableV","kind":"typeIdentifier"},{"kind":"text","text":" : "},{"text":"Any","kind":"keyword"},{"kind":"text","text":"])"}],"title":"NfinityCloudError.httpDeserialization(httpCode:cause:headers:)","abstract":[{"text":"Indicates a problem with deserializing the HTTP response into a model object.","type":"text"}],"role":"symbol","kind":"symbol"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework":{"role":"collection","identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework","url":"\/documentation\/passwordandnfinityframework","title":"PasswordAndNfinityFramework","kind":"symbol","type":"topic","abstract":[]},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError":{"kind":"symbol","role":"symbol","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror","abstract":[{"text":"Represents an enum hierarchy for errors related to the Nfinity Cloud.","type":"text"}],"fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"NfinityCloudError"}],"title":"NfinityCloudError","navigatorTitle":[{"kind":"identifier","text":"NfinityCloudError"}],"type":"topic","identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/FaceDetectionSender/sendFaceDetection(imageData:)":{"abstract":[{"text":"Method for sending face detection data, and returns an ","type":"text"},{"type":"codeVoice","code":"Swift.Result"},{"text":" instance.","type":"text"}],"type":"topic","role":"symbol","kind":"symbol","url":"\/documentation\/passwordandnfinityframework\/facedetectionsender\/sendfacedetection(imagedata:)","title":"sendFaceDetection(imageData:)","identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/FaceDetectionSender\/sendFaceDetection(imageData:)","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"sendFaceDetection"},{"kind":"text","text":"("},{"kind":"externalParam","text":"imageData"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:10Foundation4DataV","kind":"typeIdentifier","text":"Data"},{"kind":"text","text":") "},{"kind":"keyword","text":"async"},{"text":" -> Swift","kind":"text"},{"kind":"text","text":"."},{"text":"Result","kind":"typeIdentifier","preciseIdentifier":"s:s6ResultO"},{"text":"<","kind":"text"},{"kind":"typeIdentifier","text":"FaceDetectionState","preciseIdentifier":"s:27PasswordAndNfinityFramework18FaceDetectionStateO"},{"kind":"text","text":", "},{"text":"NfinityCloudError","preciseIdentifier":"s:27PasswordAndNfinityFramework0C10CloudErrorO","kind":"typeIdentifier"},{"kind":"text","text":">"}]},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError/cloudError(errorCode:errorMessage:)":{"identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/cloudError(errorCode:errorMessage:)","type":"topic","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror\/clouderror(errorcode:errormessage:)","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"kind":"identifier","text":"cloudError"},{"text":"(","kind":"text"},{"text":"errorCode","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":"?, "},{"text":"errorMessage","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"String","preciseIdentifier":"s:SS"},{"kind":"text","text":"?)"}],"abstract":[{"text":"Represents an error with a specific error code and message that the Nfinity Cloud might return, but not from network call.","type":"text"}],"title":"NfinityCloudError.cloudError(errorCode:errorMessage:)","role":"symbol","kind":"symbol"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError/network(cause:)":{"identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/network(cause:)","type":"topic","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror\/network(cause:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"text":"network","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"cause"},{"kind":"text","text":": "},{"preciseIdentifier":"s:21CommunicatorFramework0A5ErrorO","text":"CommunicatorError","kind":"typeIdentifier"},{"text":")","kind":"text"}],"title":"NfinityCloudError.network(cause:)","abstract":[{"text":"Case represents network error","type":"text"}],"role":"symbol","kind":"symbol"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError/Error-Implementations":{"type":"topic","role":"collectionGroup","identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/Error-Implementations","title":"Error Implementations","abstract":[],"kind":"article","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror\/error-implementations"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError/cloudHTTPError(httpCode:response:)":{"identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/cloudHTTPError(httpCode:response:)","kind":"symbol","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"text":"cloudHTTPError","kind":"identifier"},{"kind":"text","text":"("},{"text":"httpCode","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"Int","preciseIdentifier":"s:Si"},{"kind":"text","text":", "},{"text":"response","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"NfinityCloudErrorResponse","preciseIdentifier":"s:27PasswordAndNfinityFramework0C18CloudErrorResponseV"},{"text":")","kind":"text"}],"title":"NfinityCloudError.cloudHTTPError(httpCode:response:)","abstract":[{"type":"text","text":"Represents an HTTP error from the Nfinity Cloud with the associated HTTP status code and a custom response object."}],"role":"symbol","type":"topic","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror\/cloudhttperror(httpcode:response:)"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/s8SendableP":{"title":"Swift.Sendable","identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/s8SendableP","type":"unresolvable"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/s5ErrorP":{"title":"Swift.Error","identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/s5ErrorP","type":"unresolvable"},"doc://com.aheaditec.widgetizing.PasswordAndNfinityFramework/documentation/PasswordAndNfinityFramework/NfinityCloudError/attemptExceeded":{"identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/NfinityCloudError\/attemptExceeded","type":"topic","url":"\/documentation\/passwordandnfinityframework\/nfinityclouderror\/attemptexceeded","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"text":"attemptExceeded","kind":"identifier"}],"abstract":[{"text":"Case represents too many tries on ","type":"text"},{"identifier":"doc:\/\/com.aheaditec.widgetizing.PasswordAndNfinityFramework\/documentation\/PasswordAndNfinityFramework\/FaceDetectionSender\/sendFaceDetection(imageData:)","type":"reference","isActive":true},{"type":"text","text":"."}],"title":"NfinityCloudError.attemptExceeded","role":"symbol","kind":"symbol"}}}